restapi/src/main/java/io/stargate/web/resources/RowResource.java

```diff
@@ -546,7 +558,8 @@ public Response updateRow(
       final RowUpdate changeSet) {
     return RequestHandler.handle(
         () -> {
-          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token);
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
```

Comment:
This is problematic. It calls the `authenticationService.validateToken(token, headers);` that in turn call the:
```
default AuthenticationSubject validateToken(String token, Map<String, String> headers)
      throws UnauthorizedException {
    return validateToken(token);
  }
```
and headers are ignored, and not propagated. Similar for other calls

Reply:
Authentication implementations that need to examine headers are expected to override `validateToken(String token, Map<String, String> headers)`

Reply:
the `default` method is a convenience for "simple" authenticators